clustoapi/server.py

```diff
@@ -464,6 +504,104 @@ def get_by_names():
     return util.dumps(objs, 200 if all(objs) else 206 if any(objs) else 404)
 
 
+@root_app.get('/by-attr')
+def get_by_attr():
+    """
+One of the main ``clusto`` operations. Parameters:
+
+* Required: the ``key`` parameter
+* Optional: the ``subkey`` parameter
+* Optional: the ``value`` parameter
+
+Examples:
+
+.. code:: bash
+
+    $ ${get} ${server_url}/by-attr
+    "Provide a key to use get_by_attr"
+    HTTP: 412
+    Content-type: application/json
+
+    $ ${get} -d 'key=nonkey' ${server_url}/by-attr
+    []
+    HTTP: 200
```

Comment:
We talked about this. Return value shouldn't be >299 because core clusto returns an empty list.
```
clusto [1]> get_by_attr(key='nokey')
   out [1]> []

clusto [2]> get_from_pools(['nocollide1','nocollide2'])                                                                                                                                                                        
   out [2]> set()

clusto [3]> get_by_name('noname')
LookupError: noname does not exist.
```